File: docs/snippets/Datamaps/attach-datamaps/attach-to-assay.py
# docs:begin
from arctrl import ArcAssay, DataContext, Datamap

leaf_area = DataContext(
    name="assays/Measurement/dataset/results.csv",
    label="leaf area",
)

datamap = Datamap([leaf_area])

assay = ArcAssay("Measurement")

assay.Datamap = datamap
# docs:end

# docs:assert
if assay.Datamap is None:
    raise Exception("Expected datamap to be attached to the assay.")

if len(assay.Datamap.DataContexts) != 1:
    raise Exception(
        f"Expected one attached data context, got {len(assay.Datamap.DataContexts)}"
    )
# docs:endassert

File: docs/snippets/Datamaps/attach-datamaps/attach-to-assay.ts
// docs:begin
import {
  ArcAssay,
  DataContext,
  Datamap,
} from "@nfdi4plants/arctrl";

const leafArea = new DataContext(
  undefined,
  "assays/Measurement/dataset/results.csv",
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  "leaf area",
);

const datamap = new Datamap([leafArea]);

const assay = new ArcAssay("Measurement");

assay.Datamap = datamap;
// docs:end

// docs:assert
if (assay.Datamap === undefined) {
  throw new Error("Expected datamap to be attached to the assay.");
}

if (assay.Datamap.DataContexts.length !== 1) {
  throw new Error(
    `Expected one attached data context, got ${assay.Datamap.DataContexts.length}`,
  );
}
// docs:endassert
